pkgs/applications/science/misc/root/default.nix

```diff
@@ -178,6 +195,54 @@ stdenv.mkDerivation rec {
         --set PYTHONPATH "$out/lib" \
         --set ${lib.optionalString stdenv.isDarwin "DY"}LD_LIBRARY_PATH "$out/lib"
     done
+
+    # Make ldd and sed available to the ROOT executable
+    wrapProgram "$out/bin/root" --prefix PATH : "${lib.makeBinPath [ stdenv.cc.libc gnused ]}"
+
+    # Patch thisroot.{sh,csh,fish} setxrd.sh
+    patchCommandInplace(){
+      local file="$1"
+      shift
+      local sedCommand=""
+      local arg
+      while [ "$#" -gt 0 ]; do
+        local exeName="$1"
+        local exePath="$2"
+        shift 2
+        sedCommand="$sedCommand s|^$exeName\$|$exePath|g;"
+        sedCommand="$sedCommand s|^$exeName\\([^A-Za-z_-]\\)|$exePath\\1|g;"
+        sedCommand="$sedCommand s|\\([^A-Za-z_-]\\)$exeName\$|\\1$exePath|g;"
+        sedCommand="$sedCommand s|\\([^A-Za-z_-]\\)$exeName\\([^A-Za-z_-]\\)|\\1$exePath\\2|g;"
+      done
+      sed -i "''${sedCommand:1}" "$file"
+    }
+    patchWhichInplace(){
+      for file in "$@"; do
+        patchCommandInplace "$file" \
+          "which -p" which-p \
+          which "${which}/bin/which" \
+          which-p "which -p"
+      done
+    }
+    patchCommandInplace "$out/bin/thisroot.sh" \
+      "grep" "${gnugrep}/bin/grep" \
+      "manpath" "${man}/bin/manpath" \
+      "man" "${man}/bin/man" \
+      "ps" "${procps}/bin/ps" \
+      "sed" "${gnused}/bin/sed"
```

Comment:
Is there a reason why
```bash
substituteInPlace "$out/bin/thisroot.sh" \
  --replace "grep" "${gnugrep}/bin/grep" \
  # ...
```
won't work?

Reply:
As a general approach, direct substitution may cause issue of replacing longer words that contains the sub-string. For example, `manpath` might be affected by the substitution of `man`.

Reply:
That is why you substitute "manpath" first?

Reply:
@veprbl I didn't thought about the order (which is roughly alphabetic) actually. In the `patchCommandInplace` bash function, I use four sed commands to simulate the regex lookaround and ensure that the whole word is being replaced.

Reply:
Alternative idea: put all this stuff to `propagatedBuildInputs`